### PR TITLE
Add Rubocop (Part I)

### PR DESCRIPTION
**Pull request by amirmujkic**
- Added rubocop and rubocop-rspec
- Fixed rubocop offenses
- Added .rubocop_todo.yml -> TODO in Part II

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,13 +11,13 @@ gemspec
...
 group :mysql, optional: true do
-  gem "mysql2", "~> 0.5.2"
+  gem 'mysql2', '~> 0.5.2'
 end
 
 group :sqlserver, optional: true do
-  gem "tiny_tds", "~> 2.1.2"
-  gem "activerecord-sqlserver-adapter", "~> 5.1.6"
+  gem 'activerecord-sqlserver-adapter', '~> 5.1.6'
+  gem 'tiny_tds', '~> 2.1.2'
```
Comment (klausmeyer) [inline Gemfile]: I'm not 100% sure but I think you'll need to keep `tiny_tds` loaded before `activerecord-sqlserver-adapter`.
Could be related to the failing sqlserver CI jobs.

https://github.com/rails-sqlserver/activerecord-sqlserver-adapter#installation

Comment (klausmeyer) [reply Gemfile]: Hm ... obviously I was wrong about that (since CI is red again). Sorry 😓

Comment (mkon) [reply Gemfile]: You should not rely on the order of gems in the gemspec, bundler maybe does not even load them in the order of appearance anyway. If order is important you should always do manual `require` calls. Usually when building gems all dependencies have to be manually required anyway.

Comment (amirmujkic) [reply Gemfile]: I actually did not know this, but it seems that bundler loads them in the order of appearance according to [this](https://github.com/bundler/bundler/commit/f17183b53edc96162cd2e20255d6bc315779b5cf)

**Pull request merged**